js/Common.js

```diff
@@ -809,7 +809,20 @@ if (!CRM.vars) CRM.vars = {};
       return '';
     }
     if (createLinks === true) {
-      createLinks = params.contact_type ? _.where(CRM.config.entityRef.links[entity], {type: params.contact_type}) : CRM.config.entityRef.links[entity];
+      if (!params.contact_type) {
+        createLinks = CRM.config.entityRef.links[entity];
+      }
+      else if (typeof params.contact_type === 'string') {
+        createLinks = _.where(CRM.config.entityRef.links[entity], {type: params.contact_type});
+      } else {
+        // lets assume it's an array with filters such as IN etc
+        createLinks = [];
+        _.each(params.contact_type, function(types) {
+          _.each(types, function(type) {
+            createLinks.push(_.findWhere(CRM.config.entityRef.links[entity], {type: type}));
```

Comment:
I think this is ok because it seems pre-existing that filters other than `=` and `IN` don't seem to work anyway. What I mean is e.g. if I do `'api' => ['params' => ['contact_type' => ['NOT IN' => ['Household', 'Organization']]]]`, then the widget doesn't return any results for anything, but the above code adds "New" links for Household and Org. But it also returns no results before the patch, unrelated to links (seems like an api3 problem?).

Reply:
Yes, I didn't want to hard-code for `IN` hence the patch.